src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -1571,32 +960,87 @@ where
     ((ok_streams, err_stream), None)
 }
 
+/// Reclocks an `IntoTime` frontier stream into a `FromTime` frontier stream. This is used for the
+/// feedback edge that converts the `IntoTime` resumption frontier into the `FromTime` frontier
+/// that is used with the source's `OffsetCommiter`.
+fn reclock_resume_upper<FromTime, IntoTime>(
```

Comment:
this is MUCH nicer than doing it inline in the source reader operator!!

```suggestion
/// Reclocks an `IntoTime` frontier stream into a `FromTime` frontier stream. This is used for the
/// virtual (through persist) feedback edge so that we convert the `IntoTime` resumption frontier into the `FromTime` frontier 
/// that is used with the source's `OffsetCommiter`.
///
/// Note that we also use this async-`Stream` converter as a convenient place to compact the `ReclockFollower` trace that is currently shared between this and the `reclock_operator`.
fn reclock_resume_upper<FromTime, IntoTime>(
```